includes/compat.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * ActivityPub implementation for WordPress/PHP functions either missing from older WordPress/PHP versions or not included by default.
+ */
+
+if ( ! function_exists( 'str_starts_with' ) ) {
+	/**
+	 * Polyfill for `str_starts_with()` function added in PHP 8.0.
+	 *
+	 * Performs a case-sensitive check indicating if
+	 * the haystack begins with needle.
+	 *
+	 * @param string $haystack The string to search in.
+	 * @param string $needle   The substring to search for in the `$haystack`.
+	 * @return bool True if `$haystack` starts with `$needle`, otherwise false.
+	 */
+	function str_starts_with( $haystack, $needle ) {
```

Comment:
Instead of adding this polyfill, would it be an option to start requiring WordPress 5.9.0 to use the ActivityPub plugin? WP 5.9 was released 2 years ago now, and includes this polyfill. Maybe it would be fair to expect folks to run WP 5.9 or newer by now?

Reply:
Hmmm...

I wouldn't generally stick with older versions, and if there's a feature we really need, it makes perfect sense to increase the requirements, but in this case, and if it's so easy to maintain backwards compatibility, I wouldn't force users to have a newer version.